controllers/goflowkube/goflowkube_objects.go

```diff
@@ -183,21 +182,29 @@ func buildConfigMap(desiredGoflowKube *flowsv1alpha1.FlowCollectorGoflowKube,
 	return &configMap, digest
 }
 
-func buildService(desired *flowsv1alpha1.FlowCollectorGoflowKube, ns string) *corev1.Service {
-	return &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      constants.GoflowKubeName,
-			Namespace: ns,
-			Labels:    buildLabels(),
-		},
-		Spec: corev1.ServiceSpec{
-			Selector: buildLabels(),
-			Ports: []corev1.ServicePort{{
-				Port:     desired.Port,
-				Protocol: corev1.ProtocolUDP,
-			}},
-		},
+func buildService(old *corev1.Service, desired *flowsv1alpha1.FlowCollectorGoflowKube, ns string) *corev1.Service {
+	if old == nil {
+		return &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.GoflowKubeName,
+				Namespace: ns,
+				Labels:    buildLabels(),
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: buildLabels(),
+				Ports: []corev1.ServicePort{{
+					Port:     desired.Port,
+					Protocol: corev1.ProtocolUDP,
+				}},
+			},
+		}
 	}
+	// In case we're updating an existing service, we need to build from the old one to keep immutable fields such as clusterIP
```

Comment:
Is it safe to directly modify the old one ? Can we clone it before any modifications ?

As a user of the buildService function I would not expect the old one to be modified.

Reply:
good point, it's amended